src/nvim/message.c

```diff
@@ -3175,17 +3173,18 @@ int vim_vsnprintf(char *str, size_t str_m, char *fmt, va_list ap, typval_T *tvs)
       // parse field width
       if (*p == '*') {
         p++;
-        int j = tvs ? tv_nr(tvs, &arg_idx) : va_arg(ap, int);
-        if (j >= 0)
-          min_field_width = j;
-        else {
-          min_field_width = -j;
+        long j = tvs ? tv_nr(tvs, &arg_idx) : va_arg(ap, int);
+        if (j >= 0) {
+          min_field_width = (size_t)j;
+        } else {
+          min_field_width = (size_t)(-j);
```

Comment:
Casting a negative value to size_t: this is a Vim bug, reported here: https://github.com/vim/vim/issues/523

You can leave this as-is in this PR.